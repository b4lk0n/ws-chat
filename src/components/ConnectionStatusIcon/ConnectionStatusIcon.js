import React from 'react';
import { bool, string } from 'prop-types';
import { isBoolean } from 'lodash-es';
import cx from 'classnames';

import styles from './ConnectionStatusIcon.module.css';

export const ConnectionStatusIcon = ({ success, className }) => (
  <span
    className={cx(
      styles.root,
      {
        [styles.success]: isBoolean(success) && success,
        [styles.failure]: isBoolean(success) && !success,
      },
      className
    )}
  />
);
ConnectionStatusIcon.propTypes = {
  success: bool,
  className: string,
};
