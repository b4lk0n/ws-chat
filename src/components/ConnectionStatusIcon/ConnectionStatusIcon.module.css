.root {
  --color: var(--surface);

  display: inline-block;
  width: 16px;
  height: 16px;
  border: 1px solid var(--outline);
  border-radius: 50%;
  background-color: var(--color);
}

.success {
  --color: var(--success);
}

.failure {
  --color: var(--error);
}
