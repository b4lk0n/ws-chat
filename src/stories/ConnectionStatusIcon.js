import React from 'react';
import { withKnobs, select } from '@storybook/addon-knobs';
import { Icon, ConnectionStatusIcon } from '../components';

export default {
  title: 'ConnectionStatusIcon',
  decorators: [withKnobs],
};

export const Default = () => (
  <Icon
    value={
      <ConnectionStatusIcon
        success={select('success', {
          '—': undefined,
          true: true,
          false: false,
        })}
      />
    }
    size={select('size', {
      '—': undefined,
      small: 'small',
      large: 'large',
    })}
  />
);
