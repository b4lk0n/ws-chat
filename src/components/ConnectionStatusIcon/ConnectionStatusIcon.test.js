import React from 'react';
import { render } from '@testing-library/react';
import { ConnectionStatusIcon } from './ConnectionStatusIcon';

describe('<ConnectionStatusIcon />', () => {
  it('should not add any additional classes by default', () => {
    const { container } = render(<ConnectionStatusIcon />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should add additional classes if provided', () => {
    const { container } = render(<ConnectionStatusIcon className={'animated'} />);
    expect(container.firstChild).toHaveClass('animated');
    expect(container.firstChild).toMatchSnapshot();
  });

  [true, false].forEach(success => {
    it(`should render success={${success}} status icon`, () => {
      const { container } = render(<ConnectionStatusIcon success={success} />);
      expect(container.firstChild).toHaveClass(success ? 'success' : 'failure');
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
