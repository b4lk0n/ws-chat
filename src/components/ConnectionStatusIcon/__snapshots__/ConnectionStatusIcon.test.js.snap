// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConnectionStatusIcon /> should add additional classes if provided 1`] = `
<span
  class="root animated"
/>
`;

exports[`<ConnectionStatusIcon /> should not add any additional classes by default 1`] = `
<span
  class="root"
/>
`;

exports[`<ConnectionStatusIcon /> should render success={false} status icon 1`] = `
<span
  class="root failure"
/>
`;

exports[`<ConnectionStatusIcon /> should render success={true} status icon 1`] = `
<span
  class="root success"
/>
`;
